source/extensions/filters/http/ratelimit/ratelimit_headers.cc

```diff
@@ -0,0 +1,73 @@
+#include "extensions/filters/http/ratelimit/ratelimit_headers.h"
+
+#include "common/http/header_map_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace RateLimitFilter {
+
+Http::ResponseHeaderMapPtr RateLimitHeaders::create(
+    Filters::Common::RateLimit::DescriptorStatusListPtr&& descriptor_statuses) {
+  Http::ResponseHeaderMapPtr result = Http::ResponseHeaderMapImpl::create();
+  if (!descriptor_statuses || descriptor_statuses->empty()) {
+    descriptor_statuses = nullptr;
+    return result;
+  }
+
+  absl::optional<envoy::service::ratelimit::v3::RateLimitResponse_DescriptorStatus>
+      min_remaining_limit_status;
+  std::ostringstream quotaPolicy;
```

Comment:
nit: `quota_policy`. 

Also please avoid using ostringstream. Prefer the absl::String functions instead for this type of case.